src/05-led-roulette/my-solution.md

```diff
@@ -9,99 +9,134 @@ Here's mine:
+
+    let mut delay = hal::Timer::new(p.TIMER0);
+
+    let p0 = hal::gpio::p0::Parts::new(p.GPIO);
+
+    // Initialize all rows and cols to off
+    let mut row1 = p0.p0_13.into_push_pull_output(hal::gpio::Level::Low).degrade();
+    let row2 = p0.p0_14.into_push_pull_output(hal::gpio::Level::Low).degrade();
+    let row3 = p0.p0_15.into_push_pull_output(hal::gpio::Level::Low).degrade();
+    let mut col1 = p0.p0_04.into_push_pull_output(hal::gpio::Level::High).degrade();
+    let col2 = p0.p0_05.into_push_pull_output(hal::gpio::Level::High).degrade();
+    let col3 = p0.p0_06.into_push_pull_output(hal::gpio::Level::High).degrade();
+    let col4 = p0.p0_07.into_push_pull_output(hal::gpio::Level::High).degrade();
+    let col5 = p0.p0_08.into_push_pull_output(hal::gpio::Level::High).degrade();
+    let col6 = p0.p0_09.into_push_pull_output(hal::gpio::Level::High).degrade();
+    let col7 = p0.p0_10.into_push_pull_output(hal::gpio::Level::High).degrade();
+    let col8 = p0.p0_11.into_push_pull_output(hal::gpio::Level::High).degrade();
+    let col9 = p0.p0_12.into_push_pull_output(hal::gpio::Level::High).degrade();
+
+    // bring up the very first LED
+    row1.set_high().unwrap();
+    col1.set_low().unwrap();
+
+    let mut cols = [col1, col2, col3, col4, col5, col6, col7, col8, col9];
+    let mut rows = [row1, row2, row3];
 
-    let ms = 50_u8;
     loop {
-        for curr in 0..8 {
-            let next = (curr + 1) % 8;
+        let mut previous_pair = (1, 1);
+        for current_pair in COMBINATIONS.iter() {
+            delay.delay_ms(30u32);
 
-            leds[next].on();
-            delay.delay_ms(ms);
-            leds[curr].off();
-            delay.delay_ms(ms);
+            rows[current_pair.0 - 1].set_high().unwrap();
+            cols[current_pair.1 - 1].set_low().unwrap();
+
+            rows[previous_pair.0 - 1].set_low().unwrap();
+            cols[previous_pair.1 - 1].set_high().unwrap();
+
+            previous_pair = *current_pair;
         }
     }
 }
-
 ```
 
 One more thing! Check that your solution also works when compiled in "release" mode:
 
 ``` console
-$ cargo build --target thumbv7em-none-eabihf --release
+$ cargo-embed --release
 ```
 
-You can test it with this `gdb` command:
+If you want to debug your "release" mode binary you'll have to use a different GDB command:
 
 ``` console
-$ # or, you could simply call `cargo run --target thumbv7em-none-eabihf --release`
-$ arm-none-eabi-gdb target/thumbv7em-none-eabihf/release/led-roulette
-$ #                                              ~~~~~~~
+$ gdb target/thumbv6m-none-eabi/release/led-roulette
 ```
 
 Binary size is something we should always keep an eye on! How big is your solution? You can check
 that using the `size` command on the release binary:
 
 ``` console
-$ # equivalent to size target/thumbv7em-none-eabihf/debug/led-roulette
-$ cargo size --target thumbv7em-none-eabihf --bin led-roulette -- -A
+$ cargo size --bin led-roulette -- -A
```

Comment:
Has it been explained how to install `cargo-size`?

Reply:
Yes, cargo size comes as part of cargo binutils which is explained here https://github.com/rust-embedded/discovery/blob/rewrite/src/03-setup/README.md#cargo-binutils